buttplug/src/server/device/protocol/mizzzee_v2_1.rs

```diff
@@ -58,53 +64,81 @@ fn scalar_to_vector(scalar: u32) -> Vec<u8> {
 
   let scale: f32 = handle_scale(scalar as f32 / 1000.0) * 1023.0;
   let modded_scale: u16 = ((scale as u16) << 6) | 60;
-
-  let first_byte: u8 = (modded_scale >> 8) as u8;
-  let second_byte: u8 = modded_scale as u8;
+  
+  let bytes = modded_scale.swap_bytes().to_be_bytes();
 
   let mut data: Vec<u8> = Vec::new();
   data.extend_from_slice(&HEADER);
   data.extend_from_slice(&FILL_VEC);
-  data.extend_from_slice(&[second_byte, first_byte]);
+  data.extend_from_slice(&bytes);
   data.extend_from_slice(&FILL_VEC);
-  data.extend_from_slice(&[second_byte, first_byte]);
+  data.extend_from_slice(&bytes);
   data.push(0x00);
 
   data
 }
 
-async fn vibration_update_handler(device: Arc<Hardware>, command_holder: Arc<RwLock<Vec<u8>>>) {
+async fn vibration_update_handler(
+  device: Arc<Hardware>,
+  last_scalar_holder: Arc<RwLock<u32>>,
```

Comment:
Everything that's an `Arc<RwLock<T>>` here can just be a `Arc<AtomicU32>` or `Arc<AtomicU8>`. You can then use .store()/load() (please used Relaxed ordering) without worrying about locking.

Reply:
Ahhhh. Arc<RwLock<Vec>> was used because it has stored vectors instead of simple types
But I'm using simple types here so everything can be replaced with much simplier type.
Am I got you right?

Reply:
Yes: the Vec<> needs the lock wrapper to be thread safe, whilst the simple u32 (or u16) has an atomic type that's simpler (and potentially more efficient depending on CPU architecture)